### ISSUE TITLE
🐛 Panic `index out of bounds: the len is 3 but the index is 3` in `crates/biome_control_flow/src/builder.rs`

### ISSUE THREAD

**Issue by qarmin** [biomejs/biome#4723]
### Environment information

```block
CLI:
  Version:                      0.0.0 (self compiled c3c405f2f69cf6da889947a46a2444bcf70e1a25)
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           unset
  JS_RUNTIME_NAME:              unset
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       unset

Workspace:
  Open Documents:               0
```

### What happened?

File content(at the bottom should be attached raw, not formatted file - github removes some non-printable characters, so copying from here may not work):
```
   if ndpato{      (){   if asCmesnen{   }
  if     '&
  else
```

command
```
timeout -v 150 biome lint TEST___FILE.js
```

cause this
```
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: /home/runner/work/Automated-Fuzzer/Automated-Fuzzer/biome-main/crates/biome_control_flow/src/builder.rs:114:44
Thread Name: biome::worker_2
Message: index out of bounds: the len is 3 but the index is 3


##### Automatic Fuzzer note, output status "None", output signal "Some(6)"
```

[compressed.zip](https://github.com/user-attachments/files/18106253/compressed.zip)

### Expected result

No crash

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by arendjr** [comment]
Thanks  again for these excellent reports! This one will also be fixed in 2.0.
